src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/ResetUserPasswordTask.cs

```diff
@@ -0,0 +1,159 @@
+        }
+
+        // The technical name of the activity. Activities on a workflow definition reference this name.
+        public override string Name => nameof(ResetUserPasswordTask);
+
+        public override LocalizedString DisplayText => S["Reset User Password Task"];
+
+        // The category to which this activity belongs. The activity picker groups activities by this category.
+        public override LocalizedString Category => S["User"];
+
+        public WorkflowExpression<string> UserName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> ResetPasswordEmailSubject
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        // The message to display.
+        public WorkflowExpression<string> ResetPasswordEmailTemplate
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        // Returns the possible outcomes of this activity.
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Done"], S["Invalid"]);
+        }
+
+        // This is the heart of the activity and actually performs the work to be done.
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var userName = await _expressionEvaluator.EvaluateAsync(UserName, workflowContext, null);
+
+            var user = await _userService.GetForgotPasswordUserAsync(userName) as User;
+            if (user == null || ((await _siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().UsersMustValidateEmail && !await _userManager.IsEmailConfirmedAsync(user)))
+            {
+                var updater = _updateModelAccessor.ModelUpdater;
+                if (updater != null)
+                {
+                    updater.ModelState.TryAddModelError("", S["No email service is available"]);
+                }
+                return Outcomes("Invalid");
+            }
+            else
+            {
+                user.ResetToken = Convert.ToBase64String(Encoding.UTF8.GetBytes(user.ResetToken));
+                var resetPasswordUrl = _linkGenerator.GetUriByAction(_httpContextAccessor.HttpContext,
+                    "ResetPassword", "ResetPassword", new { area = "OrchardCore.Users", code = user.ResetToken });
+
+                workflowContext.Properties["ResetPasswordUrl"] = resetPasswordUrl;
+
+                var subject = await _expressionEvaluator.EvaluateAsync(ResetPasswordEmailSubject, workflowContext, null);
+                var localizedSubject = new LocalizedString(nameof(ResetUserPasswordTask), subject);
```

Comment:
Not a localized string.

Reply:
So I mostly followed the convention used in RegisterUserTask.cs, which is already in the dev branch, and it had the following lines:
```chsarp
var subject = await _expressionEvaluator.EvaluateAsync(ConfirmationEmailSubject, workflowContext, null);
                    var localizedSubject = new LocalizedString(nameof(RegisterUserTask), subject);

                    var body = await _expressionEvaluator.EvaluateAsync(ConfirmationEmailTemplate, workflowContext, _htmlEncoder);
                    var localizedBody = new LocalizedHtmlString(nameof(RegisterUserTask), body);
                    var message = new MailMessage()
                    {
                        To = email,
                        Subject = localizedSubject.ResourceNotFound ? subject : localizedSubject.Value,
                        Body = localizedBody.IsResourceNotFound ? body : localizedBody.Value,
                        IsBodyHtml = true
                    };
```

Shall I fix both classes to remove the use of localized strings ? Or is there a purpose to use them in RegisterUserTask?

Reply:
We need to fix it there too. You can remove it from your code in the meantime.